pkg/auth/auth.go

```diff
@@ -34,8 +35,22 @@ func CheckAuthFile(authfile string) error {
 }
 
 // Login login to the server with creds from Stdin or CLI
-func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginOptions, registry string) error {
-	server := getRegistryName(registry)
+func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginOptions, registry []string) error {
+	var (
+		server string
+		err    error
+	)
+	if len(registry) == 0 && !opts.AcceptUnspecifiedRegistry {
+		return errors.Errorf("please provide a registry to login to")
+	}
+	if len(registry) == 0 {
+		if server, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
+			return err
+		}
+		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", server)
+	} else {
+		server = getRegistryName(registry[0])
+	}
```

Comment:
Also add a check for `len(registry) > 1` so that the caller really has to do nothing.

Applies similarly to the logout code.